### DISCUSSION TITLE
`Database not found` error with unencrypted iOS backup

### DISCUSSION THREAD

**Discussion by RayTCosgrove** [ReagentX/imessage-exporter#359] [Q&A] [answered]
When attempting to export messages from an unencrypted iphone back up freshly created today, i get the following error:

`Invalid configuration: Database not found at ~/Library/Application Support/MobileSync/Backup/00008120-001610812260201E/3d/3d0d7e5fb2ce288813306e4d4636395e047a3d28`

I checked the 3d/ directory and the file 3d0d7e5fb2ce288813306e4d4636395e047a3d28 is not present in the backup. Perhaps the format of the backups has changed? Is there possibly a simple fix i could use to update which files are looked for in the backup?

Thanks

**Comment by ReagentX** [comment]
The backup format has not changed. Are you sure that the backup was not encrypted with a password? 

What version of iOS/MacOS are you using?

**Reply by RayTCosgrove** [reply]
yes positive, 
<img width="1320" alt="image" src="https://github.com/user-attachments/assets/70ff8884-1ae3-4cd0-baa4-f3e5e3140aba">


IPhone 14 Pro max with ios 18.2
Macbook air m3 with macOs 14.5

**Reply by ReagentX** [reply, marked as answer]
You didn't create a full backup; you need to select `Back up all of the data on your iPhone to this Mac`.

**Reply by m-a-flynn** [reply]
Thank you for creating this tool! We were running into the same problem -- "Database not found at ~/Library/Application Support/MobileSync/Backup/0000.../3d/3d0d7..." but we ensured to check "Back up all of the data on your iPhone to this Mac" and the issue persists. We ran the command "imessage-exporter -f html -p ~/[root directory of backup] -a iOS -o backup_export"

**Reply by ReagentX** [reply]
Did you make a new backup after enabling that option?

**Reply by m-a-flynn** [reply]
Yes, we ensured to hit sync and apply and tried running the command again after making sure the option was ticked

**Reply by ReagentX** [reply]
Syncing isn't enough—you have to actually make a full backup by clicking "Back Up Now" in the interface.

**Reply by potatoesocean9816432983** [reply]
Hi there. Thank you for building the app. I have struggled with this same error and have reviewed the previous posts including this one. I made four backups - encrypted and not, using iTunes and the windows store apple devices app. I still get the below. Using windows and XS on latest iOS. Tried without and without -a, as you can see. It's possible this is because I recently turned off icloud to download all my messages to the device locally. Manual inspection of db contents has not shown any obvious reason for this.
I have made sure the 'full backup' option is always selected.

`
C:\Users\user\Apple\MobileSync\Backup>imessage-exporter -f html -c clone
Invalid configuration: Database not found at /Library/Messages/chat.db

C:\Users\user\Apple\MobileSync\Backup>imessage-exporter -f html -c clone -a iOS
Invalid configuration: Database not found at /Library/Messages/chat.db\3d/3d0d7e5.....
`

**Reply by ReagentX** [reply]
You need to use the `-p` flag, not run `imessage-exporter` from inside of the backup. See the instructions and examples [here](https://github.com/ReagentX/imessage-exporter/tree/develop/imessage-exporter#how-to-use).
